### PR TITLE
Improvements to Expression Trees tutorial

### PR DESCRIPTION
**Pull request by svick**
Several improvements to the Expression Trees tutorial.

Some of the commits contain further details or rationale for the change.

CC: @BillWagner

### CODE REVIEW

**Comment by svick** [comment]
One more thing I noticed: [the Building article](https://dotnet.github.io/docs/languages/csharp/expression-trees-building.html#building-code-in-depth) talks about "the samples section". That section is not mentioned elsewhere (I think) and is not linked from the articles.

I think it should be either linked, or not mentioned at all.

File: samples/csharp-language/expression-trees/ExpressionTreeBuildingSampleOne.cs [hunk 1]
```diff
@@ -27,9 +28,8 @@ public override void Run()
             var ySquared = Expression.Multiply(yParameter, yParameter);
             var sum = Expression.Add(xSquared, ySquared);
 
-            Expression<Func<double, double>> sqrt = (x) => Math.Sqrt(x);
-            var methodCall = sqrt.Body as MethodCallExpression;
-            var distance = methodCall.Update(default(Expression), new List<Expression> { sum });
+            var sqrtMethod = typeof(Math).GetMethod("Sqrt", new[] { typeof(double) });
+            var distance = Expression.Call(sqrtMethod, sum);
```
Comment (BillWagner) [inline samples/csharp-language/expression-trees/ExpressionTreeBuildingSampleOne.cs]: I like this code better, too :smile:

Unfortunately, it's not supported on .NET Core. That's why I went with the original (and more complicated) construct.

If you can back out this one code change (and the example code in the .md file) I'll merge.

**Comment by BillWagner** [comment]
Overall, this looks great. I made once comment because of a construct that doesn't work in .NET Core. If you can back that out, I'll merge it.

**Comment by BillWagner** [comment]
@svick The text for "samples section" should link to this GitHub page: https://github.com/dotnet/core-docs/tree/master/samples/csharp-language/expression-trees

Thanks for fixing that.

File: samples/csharp-language/expression-trees/ExpressionTreeBuildingSampleOne.cs (hunk 1 above)
Comment (svick) [inline samples/csharp-language/expression-trees/ExpressionTreeBuildingSampleOne.cs]: It is supported. Except it's [an extension method in the `System.Reflection` namespace](https://dotnet.github.io/api/System.Reflection.TypeExtensions.html#System_Reflection_TypeExtensions_GetMethod_System_Type_System_String_System_Type___), which is why I also added the `using`.

**Comment by svick** [comment]
@BillWagner Ok, link added.

File: samples/csharp-language/expression-trees/ExpressionTreeBuildingSampleOne.cs (hunk 1 above)
Comment (BillWagner) [inline samples/csharp-language/expression-trees/ExpressionTreeBuildingSampleOne.cs]: Doh!  My bad. I don't know how I missed that.

**Comment by BillWagner** [comment]
@svick If you don't have any other changes, I'll merge this. Let me know.

**Comment by svick** [comment]
@BillWagner Yeah, this is all I have. Feel free to merge.

**Pull request merged**